### PR TITLE
Add precommit and update tox targets

### LINKED ISSUE #15: Tox: Pass through extra arguments to the test command
https://github.com/wagtail/wagtail-localize/pull/480/files#r740987231

### LINKED ISSUE #21: Add Black formatting by default
* Now that Black is being used in Wagtail core, and a 'must' for new Wagtail projects we should adopt in this cookiecutter also

### PR DESCRIPTION
**Pull request by zerolab**
Fixes #15
Fixes #21

the eslint job should be added after #20

### CODE REVIEW

File: wagtail-{{ cookiecutter.project_name_kebab }}/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[flake8]
+max-line-length=88
+exclude=migrations,dist,build,venv,node_modules,.tox,venv,.venv,.git
+# E501: Line too long
+# W503: line break before binary operator (superseded by W504 line break after binary operator)
+ignore = E501,W503
+
+[isort]
+known_first_party=wagtail_localize
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/setup.cfg:9]: `wagtail_{{ cookiecutter.project_name_snake }}`

File: wagtail-{{ cookiecutter.project_name_kebab }}/tox.ini [hunk 2]
```diff
@@ -2,43 +2,51 @@
...
-[flake8]
-# E501: Line too long
-# W503: line break before binary operator (superseded by W504 line break after binary operator)
-ignore = E501,W503
-exclude = migrations,node_modules
+envlist =
+    python{3.7,3.8,3.9}-django{3.2}-wagtail{2.15,3.0,4.0}-{sqlite,postgres}
+    python{3.10}-django{3.2}-wagtail{2.15,3.0,4.0,main}-{sqlite,postgres}
+    python{3.7,3.8,3.9,3.10}-django{4.0}-wagtail{3.0,4.0,main}-{sqlite,postgres}
+    python{3.8,3.9,3.10}-django{4.1,main}-wagtail{4.0,main}-{sqlite,postgres}
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/tox.ini:9]: Python 3.11 has been released in the time between submitting this and someone reviewing it, and wagtail 4.1. Shall we get both tox and the workflow file testing on the latest versions, or just get this merged and then do another Pr bumping them. I don't want to let perfect be the enemy of the good too much here..

File: wagtail-{{ cookiecutter.project_name_kebab }}/README.md [hunk 3]
```diff
@@ -34,3 +34,52 @@
...
+
+
+## Contributing
+
+To make changes to this project, first clone this repository:
+
+```sh
+git clone git@https://github.com/wagtail/{{ cookiecutter.project_name_kebab }}.git
+cd {{ cookiecutter.project_name_kebab }}
+```
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/README.md:46]:
```
git clone git@https://github.com/{{ cookiecutter.github_username }}/wagtail-{{ cookiecutter.project_name_kebab }}.git
cd wagtail-{{ cookiecutter.project_name_kebab }}
```

Comment (chris48s) [reply wagtail-{{ cookiecutter.project_name_kebab }}/README.md:46]: Side note: I'd quite like to make some variables in the cookiecutter setup that include the `wagtail-` prefix but I'd like to try and get some of this open stuff picked off first before I shave that specific yak

File: wagtail-{{ cookiecutter.project_name_kebab }}/README.md [hunk 4]
```diff
@@ -34,3 +34,52 @@
...
+
+### pre-commit
+
+Note that this project uses [pre-commit](https://github.com/pre-commit/pre-commit). To set up locally:
+
+```shell
+# if you don't have it yet, globally
+$ pip install pre-commit
+# go to the project directory
+$ cd {{ cookiecutter.project_name_kebab }}
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/README.md:66]: `cd wagtail-{{ cookiecutter.project_name_kebab }}`

File: wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml [hunk 5]
```diff
@@ -8,24 +8,50 @@ on:
 
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml:16]: Some comment as https://github.com/wagtail/cookiecutter-wagtail-package/pull/29#discussion_r1038042000 here. This is now the default

File: wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml [hunk 6]
```diff
@@ -34,29 +60,43 @@ jobs:
 
   test-postgres:
     runs-on: ubuntu-latest
+    needs: lint
     strategy:
       matrix:
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml:65]: similar comments about python 3.11/wagtail 4.1 here

**Review by chris48s — COMMENTED** [review]
This is going to have mad merge conflicts with https://github.com/wagtail/cookiecutter-wagtail-package/pull/32 but I reckon lets get this one finished, merge it roughly as it stands, and then I'll rebase my branch onto this, update all the pip/setuptools stuff to flit.

One of your finest reviews would be appreciated in return :pray: 

If you want to kick the can of python 3.11/wagtail 4.1 down the road, I reckon that's reasonable.

File: wagtail-{{ cookiecutter.project_name_kebab }}/tox.ini (hunk 2 above)
Comment (zerolab) [reply wagtail-{{ cookiecutter.project_name_kebab }}/tox.ini]: added to the mix

File: wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml (hunk 5 above)
Comment (zerolab) [reply wagtail-{{ cookiecutter.project_name_kebab }}/.github/workflows/test.yml:16]: Leaving it there on the "explicit > implicit"

File: wagtail-{{ cookiecutter.project_name_kebab }}/README.md [hunk 7]
```diff
@@ -34,3 +34,52 @@
...
 - `pip install {{ cookiecutter.project_name_kebab }}`
 - ...
+
+
+## Contributing
+
+To make changes to this project, first clone this repository:
+
+```sh
+git clone git@https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_name_kebab }}.git
```
Comment (chris48s) [inline wagtail-{{ cookiecutter.project_name_kebab }}/README.md:44]: 1 more

```suggestion
git clone git@https://github.com/{{ cookiecutter.github_username }}/wagtail-{{ cookiecutter.project_name_kebab }}.git
```

**Comment by chris48s** [comment]
Ta. I'm going to merge this one now. I will deal with the monster rebase on https://github.com/wagtail/cookiecutter-wagtail-package/pull/32 sometime after Christmas.

**Comment by lb-** [comment]
Thanks @zerolab and all - this is a nice improvement
